src/metabase/api/dashboard.clj

```diff
@@ -997,6 +997,20 @@
                                               1 first)))
          :has_more_values has_more_values}))))
 
+(defn- combine-chained-fitler-results
+  [results]
+  (let [;; merge values with remapped values taking priority
+        values (->> (mapcat :values results)
+                    (sort-by count)
+                    (m/index-by first)
+                    vals)]
+    (cond->> values
+      (seq values)
+      ;; sort by remapped values only if all values are remapped
```

Comment:
is this accurate if it's only looking at the first value? the comment sounds like it looks at every value.

Reply:
Something like `(sort-by #(or (second %) (first %))` might be closer to the behavior described. i.e. `Sort by remapped value if it exists`

Reply:
that's not the same - as we sort the rows the first value would have 2 items only if all are remapped. Your code would do it differently